### PR TITLE
Add shadow under compose view on scroll

### LINKED ISSUE #5098: Shadow above the message text box when texts are covered
This is how Signal currently looks when texts are covered:

This is what Google messenger does when texts are covered:

As you can see, a shadow appears on the bottom of the message area on Google messenger, and not on Signal.

This bottom shadow serves two purposes: to visually communicate the presence of later messages and to provide consistency with the shadow on the action bar. It would be great if this could be added to Signal.

### PR DESCRIPTION
**Pull request by mateoeh**
### Contributor checklist
- [X] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [X] I have tested my contribution on these devices:
 * Samsung Galaxy S5, Android 6.0.1 (CyanogenMod 13)
 * AVD Nexus 5X, Android 6.0.1
 * AVD Nexus One, Android 2.3
- [X] My contribution is fully baked and ready to be merged as is
- [X] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

----------

### Description

Adds a shadow under the compose view (like the one in Google Messenger) that divides it from the conversation list when scrolling.

Fixes #5098

screenshot_20160719-202632

// FREEBIE

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 1]
```diff
@@ -104,6 +105,32 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle bu
...
+          } else if (Build.VERSION.SDK_INT > Build.VERSION_CODES.HONEYCOMB) {
+            composeDivider.setAlpha(isAtBottom() ? 0 : 1);
+          } else {
+            composeDivider.setVisibility(isAtBottom() ? View.INVISIBLE : View.VISIBLE);
+          }
+
+          wasAtBottom = isAtBottom();
+        }
+      }
+    });
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: Let's not make this anonymous if we're going to maintaining instance state, that way we can contain the helper method as well.

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 2]
```diff
@@ -104,6 +105,32 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle bu
     list.setHasFixedSize(false);
     list.setLayoutManager(layoutManager);
 
+    composeDivider = view.findViewById(R.id.compose_divider);
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
+      composeDivider.setVisibility(View.INVISIBLE);
+    }
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: why is this necessary?

**Comment by moxie0** [comment]
lookin good

**Comment by haffenloher** [comment]
Isn't there already a very subtle shadow beneath the action bar at the top? Compared to that, I feel like the shadow you added above the compose text field is too strong.

File: src/org/thoughtcrime/securesms/ConversationFragment.java (hunk 2 above)
Comment (mateoeh) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: The `alpha` attribute on the divider view does nothing in API 10 and below; `visibility` is only manipulated in API 10 and below to avoid redundancy.

**Comment by mateoeh** [comment]
@haffenloher Good point, I'll be sure to change that. Closing and resubmitting later
